src/main/ruby/core/truffle/exception_operations.rb

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+# code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+
+module Truffle
+  module ExceptionOperations
+    class Formatter
+      def initialize(format_proc)
+        @format_proc = format_proc
+      end
+
+      def format_exception(exception)
+        @format_proc.call(exception)
+      end
+    end
+    
+    NO_METHOD_ERROR = Formatter.new(
+      Proc.new do |exception|
+        if exception.receiver.respond_to?(:inspect)
+          format('undefined method %s for %s:%s', exception.name, exception.receiver, exception.receiver.class.name)
```

Comment:
You likely need `Rubinius::Type.object_class` instead of `.class` here, so that it works for BasicObject too, and doesn't get influenced if a method `class` is defined.